### DISCUSSION TITLE
What frameworks could I use to build a review website written in HTML, CSS, JavaScript, MySQL, and PHP?

### DISCUSSION THREAD

**Discussion by emullallphotography** [community/community#60893] [Programming Help]
### Body

Also, what resources could I use to learn these frameworks? I’m already using free code camp and W3schools.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by AbdrrahimDahmani** [comment]
I would suggest you to work with react it's simple and has a wide documentation. Talking about backend you can use PHP if you want to, it can be combined with react you should just run it as a server, check some Youtube tutorials about integrating PHP with React. BTW PHP is still used widely in modern web development, has not died yet you can always use it, but if you want to try some php framework try to work with symphony or Laravel.

Concerning the learning path, I would always recommend reading the documentation and learning trough practicing. By practicing you confront errors that require you to search, and by searching you learn more things

**Comment by the-officialjosh** [comment]
If I was to advise in a case where you don't have to say any experience I will say use UDEMY

I took Jonas Schmedtmann JavaScript course there and that might be enough to get you going on the PAID front end part and with the right timing you can even get the course like I did under $10. Now the rest of the HTML and CSS part are things you can learn from YouTube and now you even have ChatGPT to help you with basic structuring and CSS. You might not be able to rely on it for the JS part. Now with this and what you've learnt from W3school and FreeCodeCamp, you are solid for the frontend part.

For SQL and PHP though I am not proficient there but I am 100% that Youtube has more than enough tools to help you there. So what you need to do is just to learn the frontend build it and start learning the backend to make it work. 

Not to overwhelm you with tutorials but I hope this should guide you on this project. Good Luck

**Reply by GhostOf0days** [reply]
I don't think OP should focus on PHP even if some people use it. Instead, he should use more modern technologies. Also, [this](https://csx.codesmith.io/) is better for JavaScript and it's free. Jonas' React course is good, however. If OP decides to take the Udemy route, this is a good path, but it's expensive: https://www.udemy.com/course/the-ultimate-react-course/ -> https://www.udemy.com/course/complete-nodejs-developer-zero-to-mastery/ -> https://www.udemy.com/course/microservices-with-node-js-and-react/. Also, OP should take the free JavaScript course I recommended prior and also learn HTML and CSS prior.

**Reply by GhostOf0days** [reply]
For this path, I would recommend waiting for Udemy sales to buy courses. They get lowered to around $20.

**Comment by ghost** [comment]
@GhostOf0days why don’t you recommend learning PHP?

**Reply by GhostOf0days** [reply]
It is an old technology and used by very few companies nowadays. You should go down the JavaScript and TypeScript route. You could also do Python for backend in addition. However, using PHP for modern development will put you behind even if some people still use it.

**Reply by ghost** [reply]
Okay, i’ll use Python for backend and Node.JS for server side because both or newer technologies

**Comment by kartikxisk** [comment]
To build a review website using HTML, CSS, JavaScript, MySQL, and PHP, you have several frameworks and libraries at your disposal. These tools can help streamline your development process and provide various functionalities. Here's a list of frameworks you could consider:

1. **Front-End Frameworks:**
   - **Bootstrap:** A widely used HTML, CSS, and JavaScript framework that provides responsive design components and a grid system.
   - **Foundation:** Another responsive front-end framework with a focus on mobile-first development.
   - **Semantic UI:** A framework that emphasizes clear HTML semantics and provides a range of UI components.

2. **JavaScript Frameworks:**
   - **React:** A popular library for building user interfaces, offering component-based development and efficient DOM handling.
   - **Vue.js:** A progressive JavaScript framework for building interactive web interfaces.
   - **Angular:** A comprehensive framework for building dynamic web applications with two-way data binding.

3. **Back-End Frameworks:**
   - **Laravel:** A PHP framework known for its elegant syntax and feature-rich development environment.
   - **Symfony:** A PHP framework that follows best practices and provides various tools for building robust applications.
   - **CodeIgniter:** A lightweight PHP framework that's easy to learn and suitable for smaller projects.

4. **Database Libraries:**
   - **PDO (PHP Data Objects):** Not a framework, but a PHP extension for working with databases, including MySQL. It provides a consistent and secure way to interact with databases.

Remember that the choice of frameworks depends on your familiarity with them, the project's requirements, and your personal preferences. Also, make sure to properly sanitize and validate user inputs to prevent security vulnerabilities like SQL injection.

Your tech stack could look something like this:

- Front-end: HTML, CSS, JavaScript (with a chosen framework)
- Back-end: PHP (with a chosen framework like Laravel)
- Database: MySQL (interacted with using PDO or an ORM provided by the chosen PHP framework)

Lastly, consider using version control tools like Git and platforms like GitHub or GitLab to manage your codebase collaboratively.
